internal/gps/source_manager.go

```diff
@@ -142,8 +177,9 @@ var _ SourceManager = &SourceMgr{}
 
 // SourceManagerConfig holds configuration information for creating SourceMgrs.
 type SourceManagerConfig struct {
-	Cachedir string      // Where to store local instances of upstream sources.
-	Logger   *log.Logger // Optional info/warn logger. Discards if nil.
+	Cachedir       string      // Where to store local instances of upstream sources.
+	Logger         *log.Logger // Optional info/warn logger. Discards if nil.
+	DisableLocking bool        // True if dep SourceManager shoud NOT lock.
```

Comment:
"shoud" typo here. also, let's rewrite this a bit, too:

"True if the SourceManager should NOT use a file lock to protect the Cachedir from multiple processes."